In [8]:
import pandas as pd   # pandas helps us read and work with tables

In [17]:
df = pd.read_csv("products.csv")   # load the products into a table called df
df.head()                          # show the first few lines to check it loaded right

StockCode                  Description
0     10002   INFLATABLE POLITICAL GLOBE
1     10080     GROOVY CACTUS INFLATABLE
2     10120                 DOGGY RUBBER
3    10123C         HEARTS WRAPPING TAPE
4    10124A  SPOTS ON RED BOOKCOVER TAPE

Observations: Data has been loaded correctly!

In [20]:
def count_keyword(keyword):
    """
    Count how many product descriptions contain the given keyword.
    """
    return df['Description'].str.contains(keyword, case=False).sum()

In [21]:
fire_count    = count_keyword("Fire")
candle_count  = count_keyword("candle")
matches_count = count_keyword("matches")

print("Products with ‘Fire’:", fire_count)
print("Products with ‘candle’:", candle_count)
print("Products with ‘matches’:", matches_count)

Products with ‘Fire’: 9
Products with ‘candle’: 178
Products with ‘matches’: 6


Observations
1. We found 9 items mentioning “Fire.”
2. We found 178 items mentioning “candle.”
3. We found 6 items mentioning “matches.”

Problem Statement: One of your clients, Mr Smith, owns a general merchandise shop, and the company needs to reevaluate its current policy due to a recent surge in fire-related incidents in retail stores. The insurance company wants to understand whether Mr Smith's shop carries any items that could be potentially fire-hazardous to assess the potential risks better.

In [30]:
print("Total Hazardous items", fire_count+candle_count+matches_count)

Total Hazardous items 193


Observations: Mr Smith's shop carries 193 items that could be potentially fire-hazardous to assess the potential risks better.